### PR TITLE
Make stdin messaging docs not IPython specific

### PR DESCRIPTION
**Pull request by takluyver**
Closes jupyter/notebook#1585

### CODE REVIEW

**Comment by takluyver** [comment]
Ping @willingc for docs review

File: docs/messaging.rst [hunk 1]
```diff
@@ -1087,17 +1087,27 @@ Message type: ``clear_output``::
     The selective clearing keys are ignored in v4 and the default behavior remains the same,
     so v4 clear_output messages will be safely handled by a v4.1 frontend.
 
+.. _stdin_messages:
 
 Messages on the stdin ROUTER/DEALER sockets
 ===========================================
 
 This is a socket where the request/reply pattern goes in the opposite direction:
```
Comment (willingc) [inline docs/messaging.rst]: Perhaps to add a bit of clarity to 'this', 'its' for someone new to the architecture:

With the stdin ROUTER/DEALER socket, the request/reply pattern goes in the opposite direction of most kernel communication. With the stdin socket, the kernel makes the request, and the single frontend provides the response. This pattern allows code to prompt the user for a line of input, which reverses the typical pattern of reading from the terminal's stdin.

File: docs/messaging.rst [hunk 2]
```diff
@@ -1130,14 +1143,10 @@ When ``password`` is True, the frontend should not echo the input as it is enter
...
 
-   We do not explicitly try to forward the raw ``sys.stdin`` object, because in
-   practice the kernel should behave like an interactive program.  When a
-   program is opened on the console, the keyboard effectively takes over the
-   ``stdin`` file descriptor, and it can't be used for raw reading anymore.
-   Since the IPython kernel effectively behaves like a console program (albeit
-   one whose "keyboard" is actually living in a separate process and
-   transported over the zmq connection), raw ``stdin`` isn't expected to be
-   available.
+   This model of requesting user input is quite different from how stdin works
```
Comment (willingc) [inline docs/messaging.rst]: model -> pattern

**Comment by willingc** [comment]
@takluyver Overall, this looks good :-) I've suggested a bit of text for the intro paragraph to add clarity for someone new to the architecture. Feel free to use or leave as is.

Happy to merge this. Ping me when it's ready for merge. Thanks!

**Comment by takluyver** [comment]
Accepted most of your suggestions - the exception is that I preferred my wording at the end of the first paragraph.

**Comment by willingc** [comment]
@takluyver Totally cool. Merging. Thanks.

**Comment by gibiansky** [comment]
One more potential place: `If raw_input is called from code executed from such a frontend,
# a StdinNotImplementedError will be raised`

This is in the description of `execute_request`

**Comment by takluyver** [comment]
Thanks, see jupyter/jupyter_client#167

**Comment by gibiansky** [comment]
Thanks!

**Pull request merged**